examples/flutter_gallery/lib/demo/text_field_demo.dart

```diff
@@ -79,26 +112,19 @@ class TextFieldDemoState extends State<TextFieldDemo> {
             new Input(
               hintText: 'What do people call you?',
               labelText: 'Name',
-              formField: new FormField<String>(
-                // TODO(mpcomplete): replace with person#name=
-                setter:  (String val) { person.name = val; },
-                validator: _validateName
-              )
```

Comment:
> One problem is that the value of an Input (an InputValue) has a selection, but the data model is a String, so it can't be the exclusive ground truth.

The selection is a property of the editor, not the data model. I think as an app developer, I'd just want the framework to manage that behind the scenes and not tell me about it (in the general case).

> Another complication is with forms that you prepare and then submit (vs those that are "live"). For example the expansion panel example used to save even if you hit "Cancel"; this patch attempts to fix that.

I could see that also being fixed by providing a temp object, which gets swapped with the final object on submit. e.g. use `tempPerson` everywhere, then in onSubmit just do `finalPerson = tempPerson;`